src/Bounds.cpp

```diff
@@ -37,6 +37,45 @@ int static_sign(Expr x) {
     }
     return 0;
 }
+
+
+// Try to find a constant that is:
+// greater than or equal to the given Expr (if d is Up), or
+// less than or equal to the given Expr (if d is Down).
+enum Direction {Up, Down};
+Expr simplify_to_constant(Expr e, Direction d) {
```

Comment:
Might explicitly document that this is not guaranteed to return a constant. I might change the name as what this is really doing is trying to prove a constant bound on one side or the other and "simplify_to_constant" seems to imply a thing that gives a constant equivalent to an Expr or fails.